incubator/hnc/pkg/reconcilers/hnc_config.go

```diff
@@ -173,7 +180,23 @@ func (r *ConfigReconciler) createObjectReconciler(gvk schema.GroupVersionKind) e
 
 // SetupWithManager builds a controller with the reconciler.
 func (r *ConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	r.enqueueAffected(r.Log, "Enforce reconciliation to create a default"+
+		"HNCConfiguration singleton if it does not exist.")
```

Comment:
Also, insead of `enqueueAffected`, maybe `forceInitialReconcile`?

Reply:
Thanks for the suggestion! Changed to forceInitialReconcile() and called it after SetupWithManager() in setup.go. Also added the comments in setup.go to discuss the issue with r.Get() outside of Reconcile() that we discovered.

Reply:
> I have no _reason_ to be scared of this, but this makes me a little scared. I feel like it might be safer to enqueue something _after_ the reconciler's been started?

I might misunderstand this. Did you mean doing the enqueue after SetupWithManager() is called? That's why I moved it to setup.go and called it after SetupWithManaget().

Reply:
+1